README.md

```diff
@@ -102,4 +102,29 @@ NameVirtualHost *:443
     ProxyPass / https://127.0.0.1:8443/
     ProxyPassReverse / https://127.0.0.1:8443/
 </VirtualHost>
-~~~~
\ No newline at end of file
+~~~~
+
+API description
+--------------------
+| URL | Method | Input | Output | Description |
+| ---: |:-------- | :------ | :------- | :------ |
+| /api/info | GET |  | API version information in JSON | Information about API (welcome page), can be used to check if backend is running |
+| /api/user | GET |  | JSON with the information about the current user | Returns information about the current user |
+| /api/user/tokens | POST | Form data with the name of a token {tokenName} | JSON with the new token | Adds a new token for the current user |
+| /api/user/tokens | GET |  | JSON with the list of all user tokens | List all tokens for the current user |
+| /api//user/tokens/{tokenID} | DELETE | {tokenID} an id of a token | Server response code | Removes a specific token from the current user |
+| /api/roles | GET |  | JSON with the list of all roles | Lists all available roles |
+| /api/roles/{roleID} | GET | {roleID} an id of a role | JSON with the list of users | Returns a list of users to which a certain role was assigned |
+| /api/roles/{roleID} | POST | {roleID} an id of a role | Server response code | Assigns a specific role to the current user |
+| /api/roles/{roleID}/{userID} | DELETE | {roleID} an id of a role assigned to the user with the user id {userID} | Server response code | Removes a role from a user |
+| /api/builds | POST |  | JSON object with information about the location and build ID | Creates a temporary folder when an image has to be created. It returns a buildID identifier and a folder location. This folder is used to store a Dockerfile and files needed for the image. BuildID is a string like a UID in Java (which is generated when required and it is unique) |
+| /api/builds/{buildID} | POST | {buildID} build identifier | JSON object with information about the image and the corresponding service | Builds an image provided a buildID (which points to the folder with the Dockerfile), returns JSON with the information about the image and the new service (partly taken from the metadata) |
```

Comment:
This request requires an additional input: a multipart request body containing the files